### PR TITLE
add racr and wopr

### LINKED ISSUE #264: Add RACR and WOPR to output of `calculate_player_stats()`
RACR will be easy. 
WOPR requires team totals for target share and air yard share. 

We also need good documentation of these variables.

### PR DESCRIPTION
**Pull request by mrcaseb**
closes #264

### CODE REVIEW

**Comment by tanho63** [comment]
Possibly worthwhile to bring in PACR here? (Pass yard/pass_air_yard)

**Comment by mrcaseb** [comment]
I am not sure how to handle negative and 0 air_yards both for the weekly and the season summary. 

Here is why

``` r
# let's say a receiver played three games in 2020
# and these are his `RACR` relevant stats
df <- tibble::tribble(
  ~season, ~receiving_yards, ~receiving_air_yards,
  2020, 5, -10,
  2020, 30, 9,
  2020, 40, 1
) |> 
  dplyr::mutate(racr = receiving_yards / receiving_air_yards)

# the 40/1 game results in RACR = 40
# if that player had 1 air_yard less what's his RACR? 0? 
# Although he had 1 yard more to go?
# And how should we treat negative values? The interpretation of `RACR` is "the higher the better"
# but for negative values it is "the lower the better"
df
#> # A tibble: 3 x 4
#>   season receiving_yards receiving_air_yards  racr
#>    <dbl>           <dbl>               <dbl> <dbl>
#> 1   2020               5                 -10 -0.5 
#> 2   2020              30                   9  3.33
#> 3   2020              40                   1 40

# Now we summarise the season. What's the "correct" `RACR`?
df |> 
  dplyr::group_by(season) |> 
  dplyr::summarise(
    sum_yards = sum(receiving_yards),
    sum_air_yards = sum(receiving_air_yards),
    racr_mean = mean(racr),
    racr_from_sum = sum_yards / sum_air_yards
  )
#> # A tibble: 1 x 5
#>   season sum_yards sum_air_yards racr_mean racr_from_sum
#>    <dbl>     <dbl>         <dbl>     <dbl>         <dbl>
#> 1   2020        75             0      14.3           Inf
```

**Comment by friscojosh** [comment]
Thought I would add my thoughts here for anyone else who is ever interested.

RACR is garb for RBs be cause the interpretation of the stat gets weird when air yards are negative. If a low volume receiver has negative air yards, I would use his position as a filter to just change the air yards total to 0. I would leave RBs alone, and when talking about RACR would always point out that it is not good for fantasy analysis of their performance.

In short, if WR OR TE && air_yards <= 0 ~ airyards = 0
if RB ~ air_yards = air_yards

**Comment by mrcaseb** [comment]
The overall conclusion after I talked to Josh is

- RACR = 0 for air_yards <= 0 if the player is a Receiver
- RACR = 0 for air_yards == 0 if the player is a RB
- RACR < 0 for air_yards < 0 if the player is a RB

Probably use the Receiver definition for PACR as well. 

The positional dependency would require a roster join so I am not sure if we should just use the Receiver definition and add a note to the docs that RACR shouldn't be used for RBs

**Comment by tanho63** [comment]
I remember looking at absolute value of air yards being a decent way to fix it for negative values

-
Tan Ho
________________________________

**Comment by mrcaseb** [comment]
I think this is ready now and follows @friscojosh's `RACR` definition including RB differentiation

**Review by guga31bb — APPROVED** [review]
Looks good with the caveat that I haven't examine the output (or really used these variables before)
